src/voodoo/conversations/VoodooConversations.ts

```diff
@@ -0,0 +1,162 @@
+import { messageApi } from '@xmtp/proto'
+import { Mutex } from 'async-mutex'
+import VoodooClient, {
+  VoodooContact,
+  EncryptedVoodooMessage,
+  VoodooMessage,
+} from '../VoodooClient'
+import VoodooConversation from './VoodooConversation'
+import { utils } from '../../crypto'
+
+import {
+  buildVoodooUserInviteTopic,
+  buildVoodooDirectMessageTopic,
+} from '../utils'
+
+export default class VoodooConversations {
+  private client: VoodooClient
+  private conversations: Map<string, VoodooConversation> = new Map()
+  private convoMutex: Mutex
+
+  constructor(client: VoodooClient) {
+    this.client = client
+    this.convoMutex = new Mutex()
+  }
+
+  async list(): Promise<VoodooConversation[]> {
+    const release = await this.convoMutex.acquire()
+    try {
+      // Check for new conversations in our invite topic
+      const inviteTopic = buildVoodooUserInviteTopic(this.client.address)
+      const newConvosFromInvites = await this.client.listEnvelopes(
+        inviteTopic,
+        this.processInvite.bind(this)
+      )
+
+      for (const convo of newConvosFromInvites) {
+        if (convo) {
+          this.conversations.set(convo.peerAddress, convo)
```

Comment:
This'll break once we introduce `conversationId` support, but fine for now

Reply:
I'll make sure to call it out with a comment